dev-tool/src/list_net/data_source.rs

```diff
@@ -593,7 +595,9 @@ mod tests {
     #[test]
     fn test_data_lookup_order_changes_on_reset() {
         let mut rng = thread_rng();
-        let mut dlo = DataLookupOrder::new(vec![0, 1, 2, 3, 4]);
+        // There is a ~1.2e-46% chance that this test randomly fails
```

Comment:
Was that the reason why the android jobs failed in the past?

Reply:
No, as far as I know this only failed once on my system.

The chance of failure before had been ~0.81%, so something which will happen, but
only rarely.

The reason the android jobs failed in the past was because values becoming `NaN`.

In on case I was have to track down a situation where adding two values yielded a completely wrong result. One value was initialized to `0` and one was tested to be `0` and the outcome was something along the line of 6e15. I never was able to track down what exactly went wrong but it seemed to be related to both the `android` rustc target and qemu, and could have been a bug in either (or even both) of them.

Reply:
I see, thanks for the explanation! Would it make sense to replace `thread_rng` with a reproducible generator?

Reply:
Yes :+1: , that is a good idea.